lib/node/index.js

```diff
@@ -595,6 +598,8 @@ Request.prototype.request = function(){
     // because node will emit a faux-error "socket hang up"
     // when request is aborted before a connection is made
     if (self._aborted) return;
+    // flag retry here for out timeouts (same as above)
+    if (self._retries) return;
```

Comment:
This is also a little strange. After aborting `req` (timeout -> `abort()`), I delete the **old** `req`, but if I also `req.removeAllListeners()`, an unhandled "socket hang up" error is thrown from the **old** instance. If I don't remove listeners on the **old** `req` instance, then I have to skip somehow. 

I think the best approach would be to memoize the current retry index and return early if they are not the same.